modules/passkey/contracts/4337/Safe256BitECSignerLaunchpad.sol

```diff
@@ -0,0 +1,265 @@
+
+/**
+ * @title SafeOpLaunchpad - A contract for Safe initialization with custom unique signers that would violate ERC-4337 factory rules.
+ * @dev The is intended to be set as a Safe proxy's implementation for ERC-4337 user operation that deploys the account.
+ */
+contract Safe256BitECSignerLaunchpad is IAccount, SafeStorage, SignatureValidatorConstants {
+    bytes32 private constant DOMAIN_SEPARATOR_TYPEHASH = keccak256("EIP712Domain(uint256 chainId,address verifyingContract)");
+
+    // keccak256("SafeSignerLaunchpad.initHash") - 1
+    uint256 private constant INIT_HASH_SLOT = 0x1d2f0b9dbb6ed3f829c9614e6c5d2ea2285238801394dc57e8500e0e306d8f80;
+
+    /**
+     * @notice The keccak256 hash of the EIP-712 SafeInit struct, representing the structure of a ERC-4337 compatible deferred Safe initialization.
+     *  {address} singleton - The singleton to evolve into during the setup.
+     *  {address} signerFactory - The unique signer factory to use for creating an owner.
+     *  {bytes} signerData - The signer data to use the owner.
+     *  {address} setupTo - The contract to delegatecall during setup.
+     *  {bytes} setupData - The calldata for the setup delegatecall.
+     *  {address} fallbackHandler - The fallback handler to initialize the Safe with.
+     */
+    bytes32 private constant SAFE_INIT_TYPEHASH =
+        keccak256(
+            "SafeInit(address singleton,address signerFactory,uint256 signerX,uint256 signerY,address signerVerifier,address setupTo,bytes setupData,address fallbackHandler)"
+        );
+
+    /**
+     * @notice The keccak256 hash of the EIP-712 SafeInitOp struct, representing the user operation to execute alongside initialization.
+     *  {bytes32} userOpHash - The user operation hash being executed.
+     *  {uint48} validAfter - A timestamp representing from when the user operation is valid.
+     *  {uint48} validUntil - A timestamp representing until when the user operation is valid, or 0 to indicated "forever".
+     *  {address} entryPoint - The address of the entry point that will execute the user operation.
+     */
+    bytes32 private constant SAFE_INIT_OP_TYPEHASH =
+        keccak256("SafeInitOp(bytes32 userOpHash,uint48 validAfter,uint48 validUntil,address entryPoint)");
+
+    address private immutable SELF;
+    address public immutable SUPPORTED_ENTRYPOINT;
+
+    constructor(address entryPoint) {
+        require(entryPoint != address(0), "Invalid entry point");
+
+        SELF = address(this);
+        SUPPORTED_ENTRYPOINT = entryPoint;
+    }
+
+    modifier onlyProxy() {
+        require(singleton == SELF, "Not called from proxy");
+        _;
+    }
+
+    modifier onlySupportedEntryPoint() {
+        require(msg.sender == SUPPORTED_ENTRYPOINT, "Unsupported entry point");
+        _;
+    }
+
+    receive() external payable {}
+
+    function preValidationSetup(bytes32 initHash, address to, bytes calldata preInit) external onlyProxy {
```

Comment:
One of the reasons I had hoped to introduce this code piecemeal was that we are moving "experimental" code into the "production" directory tree, and there are quite a few things missing (for example in this file, natspec documentation, unit tests, etc).

My personal opinion is that it would have been better to do it that way just to make sure we don't overlook any of these details, which would also have kept the scope of this PR down.

Reply:
I concur. It's my bad. Since I'm going on vacation tomorrow, I wasn't sure if I would be able to add a meaningful test suite for the factory and decided to take a shortcut and migrate our existing end-to-end test, which required migrating a bunch of other experimental contracts and thus bloating the scope :slightly_frowning_face: I know that 4337 wasn't a requirement for the sprint, but I believe this migration will be beneficial in the long term because for other projects looking to use Safe with passkeys 4337 support is a must.

> we are moving "experimental" code into the "production" directory tree

I think it's okay since no code should be considered production-ready without an official release. I can also add documentation in a separate PR

Reply:
OK, Don't want to bikeshed too much given the time pressure. Can you add a GH issue to clean up some of the experimental code (like adding natspec comments, etc.) so we can schedule it for next week's sprint?

Reply:
Done here: https://github.com/safe-global/safe-modules/issues/308